mmv1/third_party/terraform/services/storage/resource_storage_bucket_object.go

```diff
@@ -376,45 +378,53 @@ func resourceStorageBucketObjectUpdate(d *schema.ResourceData, meta interface{})
 	bucket := d.Get("bucket").(string)
 	name := d.Get("name").(string)
 
-	objectsService := storage.NewObjectsService(config.NewStorageClientWithTimeoutOverride(userAgent, d.Timeout(schema.TimeoutUpdate)))
-	getCall := objectsService.Get(bucket, name)
+	if d.HasChange("content") || d.HasChange("detect_md5hash") {
```

Comment:
We'll need to include an acceptance test that covers this but also after looking at the original issue i would suggest following what Riley suggested as a solution that is more user friendly.

https://github.com/hashicorp/terraform-provider-google/issues/10488#issuecomment-1105722853

Reply:
For this one I'm not sure, because for me the update in place content must be the way it work by default not with any additionnal input neither lifecycle.
Because when you go directly on console and drag and drop a file with same name it update inplace the content, Same if you are using gcloud or gsutil.

what do you think @BBBmau  @rileykarson ?

But, agree for acceptance test I will look into it.

Reply:
Having a different UX than Console and gcloud is fine- we're a declarative tool, while gcloud is imperative, and Console has a mix of declarative-ish specifications (for things like standard updates) and imperative actions (for everything else).

Reply:
@joe-a-t ?

Reply:
> Having a different UX than Console and gcloud is fine- we're a declarative tool, while gcloud is imperative, and Console has a mix of declarative-ish specifications (for things like standard updates) and imperative actions (for everything else).

Sure it's fine, but I'm not sure it is a good and user friendly stuff to add complexity on implementation if we can make it as same as a bucket is working by design.

Reply:
furthermore with this implementation we keep the delete working as normal on resource deletion

Reply:
I added unit test for this implementation, hope it suite ;)

Reply:
@BBBmau

Reply:
@SarahFrench ;)